server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -82,8 +87,7 @@ package object server {
      * with status [[Status.NotFound]] for any requests where the function
      * is undefined.
      */
-    def apply(pf: PartialFunction[Request, Task[Response]]): HttpService =
```

Comment:
+1 I would keep this method. A partial function is essentially the most convenient way to build up routes using `dsl`.
I would also add an arg `default` to this method to give user a chance to override the default behavior (which most likely will be a `404`) if the partial function does not match a `Request`